### PR TITLE
DE- 2627: athena checker

### PR DESCRIPTION
**Pull request by setiat**
- Added new logic for path normalisation
- added schema validation for Athena.json

### CODE REVIEW

File: ids_validator/checks/generic/athena.py [hunk 1]
```diff
@@ -195,3 +192,39 @@ def get_athena_partitions_name(cls, athena_schema) -> list:
...
+        @fileId -> fileid
+        project.name -> project_name
+        """
+        normalized_path = re.sub('[^A-Za-z0-9]+', '_', path_name)
+        normalized_path = re.sub('[_]+', '_', normalized_path)
+        if normalized_path.startswith("_"):
+            normalized_path = normalized_path[1:]
+
+        if normalized_path.endswith("_"):
+            normalized_path = normalized_path[:-1]
```
Comment (mlipschultz) [inline ids_validator/checks/generic/athena.py:209]: I think these four lines could be replaced with just a `strip('_')`:
```suggestion
        normalized_path = normalized_path.strip("_")
```

File: ids_validator/utils.py [hunk 2]
```diff
@@ -18,7 +18,6 @@ def save_json(data: dict, fname):
     with open(fname, "w") as fout:
         json.dump(data, fout, indent=2, ensure_ascii=False)
 
-
```
Comment (mlipschultz) [inline ids_validator/utils.py:21]: It'd be good to keep this so it's consistent with [our coding standards](https://www.python.org/dev/peps/pep-0008/#blank-lines).

File: ids_validator/checks/generic/athena.py [hunk 3]
```diff
@@ -195,3 +192,39 @@ def get_athena_partitions_name(cls, athena_schema) -> list:
             for partition in partitions
         ]
         return names
+
+    @classmethod
+    def normalize_path_name(cls, path_name):
+        """
+        Weird-partition!@name -> weird_partition_name
```
Comment (mlipschultz) [inline ids_validator/checks/generic/athena.py:199]: Maybe I missed it, but are there tests to confirm the behavior in this method?

**Review by mlipschultz — COMMENTED** [review]
I'm mostly interested in whether there are tests to confirm the added behavior.

File: ids_validator/checks/generic/athena.py (hunk 1 above)
Comment (setiat) [reply ids_validator/checks/generic/athena.py]: Thanks! Fixed it.

File: ids_validator/utils.py (hunk 2 above)
Comment (setiat) [reply ids_validator/utils.py:21]: added 2 blank lines

File: ids_validator/checks/generic/athena.py (hunk 3 above)
Comment (setiat) [reply ids_validator/checks/generic/athena.py:199]: added test cases

File: ids_validator/utils.py (hunk 2 above)
Comment (setiat) [reply ids_validator/utils.py:21]: - Two blank lines between the import statements and other code.
- Two blank lines between each function.

right?

Comment (mlipschultz) [reply ids_validator/utils.py:21]: Right, plus two blank lines between each function or class:
> Surround top-level function and class definitions with two blank lines.

**Review by mlipschultz — APPROVED** [review]

File: ids_validator/checks/generic/athena.py [hunk 4]
```diff
@@ -195,3 +192,34 @@ def get_athena_partitions_name(cls, athena_schema) -> list:
...
+        @fileId -> fileid
+        project.name -> project_name
+        """
+        normalized_path = re.sub("[^A-Za-z0-9]+", "_", path_name)
+        normalized_path = re.sub("[_]+", "_", normalized_path)
+        normalized_path = normalized_path.strip("_")
```
Comment (mlipschultz) [inline ids_validator/checks/generic/athena.py:205]: It looks like only a leading underscore is invalid: https://github.com/tetrascience/ts-lambda-core/blob/development/lambdas/dlToAthena/normalizers/index.js#L18
```suggestion
        normalized_path = normalized_path.lstrip("_")
```

**Review by mlipschultz — CHANGES_REQUESTED** [review]

File: ids_validator/checks/generic/athena.py (hunk 4 above)
Comment (setiat) [reply ids_validator/checks/generic/athena.py]: fixed

**Review by mlipschultz — APPROVED** [review]

**Pull request merged**